dask/dataframe/io.py

```diff
@@ -626,32 +626,91 @@ def to_hdf(df, path_or_buf, key, mode='a', append=False, complevel=0,
+                if stop is None:
+                    stops.append(storer.nrows)
+                else:
+                    stops.append(stop)
+        return keys, stops
+
+    def one_path_one_key(path, key, start, stop, columns, chunksize):
+        """
+        Get the data frame corresponding to one path and one key (which should
+        not contain any wildcards).
+        """
+        if columns is None:
+            columns = list(pd.read_hdf(path, key, stop=0).columns)
 
-    token = tokenize((path_or_buf, os.path.getmtime(path_or_buf), key, start,
-                      stop, columns, chunksize))
-    name = 'read-hdf-' + token
+        token = tokenize((path, os.path.getmtime(path), key, start,
+                          stop, columns, chunksize))
+        name = 'read-hdf-' + token
 
-    dsk = dict(((name, i), (apply, pd.read_hdf,
-                             (path_or_buf, key),
-                             {'start': s, 'stop': s + chunksize,
-                              'columns': columns}))
-                for i, s in enumerate(range(start, stop, chunksize)))
+        dsk = dict(((name, i), (apply, pd.read_hdf,
+                                 (path, key),
+                                 {'start': s, 'stop': s + chunksize,
+                                  'columns': columns}))
+                    for i, s in enumerate(range(start, stop, chunksize)))
 
-    divisions = [None] * (len(dsk) + 1)
+        divisions = [None] * (len(dsk) + 1)
+        return DataFrame(dsk, name, columns, divisions)
 
-    return DataFrame(dsk, name, columns, divisions)
+    keys, stops = get_keys_and_stops(path, key, stop)
+    return concat([one_path_one_key(path, k, start, s, columns, chunksize)
+                   for k, s in zip(keys, stops)])
+
+
+@wraps(pd.read_hdf)
+def read_hdf(pattern, key, start=0, stop=None, columns=None,
+             chunksize=1000000):
+    """
+    Read hdf files into a dask dataframe. Like pandas.read_hdf, except it we
+    can read multiple files, and read multiple keys from the same file by using
+    pattern matching.
+
+    Parameters
+    ----------
+    pattern : pattern (string), or buffer to read from. Can contain wildcards
+    key : group identifier in the store. Can contain wildcards
+    start : optional, integer (defaults to 0), row number to start at
+    stop : optional, integer (defaults to None, the last row), row number to
+        stop at
+    columns : optional, a list of columns that if not None, will limit the
+        return columns
+    chunksize : optional, nrows to include in iteration, return an iterator
+
+    Returns
+    -------
+    dask.DataFrame
+    """
+    paths = sorted(glob(pattern))
+    if (start != 0 or stop is not None) and len(paths) > 1:
+        msg = ("start and stop are ambiguous for more than one file. It \n"
+        "could be interpreted as the starting and stopping index per file,\n"
+        "or starting and stopping index of the global dataset.")
```

Comment:
We need to be very clear about what's wrong, then we can motivate why we think it's wrong.